include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,126 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        /*!
+         * Allows us to avoid code duplication on the attribute creation
+         * function.
+         */
+        template < typename T >
+        static std::string type_to_string()
+        {
+            if( typeid( T ).name() == typeid( index_t ).name() )
```

Comment:
Maybe you can use std::decay to remove const, reference and so on.

Reply:
Changed it to std::is_same, which is more appropriate but was found researching std ::decay